app/views/harvests/_modal.html.haml

```diff
@@ -0,0 +1,30 @@
+#modelHarvestForm.modal.fade{"aria-hidden" => "true", "aria-labelledby" => "myModalLabel", role: "dialog", tabindex: "-1"}
+  .modal-dialog{role: "document"}
+    .modal-content
+      .modal-header.text-center
+        %h4.modal-title.w-100.font-weight-bold Record #{@crop.name} Harvest
+        %button.close{"aria-label" => "Close", "data-dismiss" => "modal", type: "button"}
+          %span{"aria-hidden" => "true"} &#215;
+      .modal-body
+        %p Select which plant part you have havested
+
+        - if @crop.plant_parts.any?
+          %h5 Recommended plant parts to harvest:
+          .index-cards
+            - @crop.plant_parts.order(:name).each do |plant_part|
+              .card
+                = link_to harvests_path(harvest: {crop_id: @crop.id, plant_part_id: plant_part.id}), method: :post, class: 'dropdown-item' do
+                  .card-title.text-center
+                    %h3= plantpart_icon(plant_part.name)
+                    %h3= plant_part.name
+
+          %hr/
+        %h6 All Plant parts
+        %nav
+          - PlantPart.all.order(:name).each do |plant_part|
```

Comment:
Would be maybe a good idea to move this to the controller? So it is clear from the controller method what data is necessary to build the view - it might improve understanding.

Reply:
If it's queried in the view, then we can cache it as part of the view (and have one less database query on the page)

Reply:
i've added caching